pydicom/pixel_data_handlers/pillow_handler.py

```diff
@@ -182,6 +183,12 @@ def get_pixeldata(ds: "Dataset") -> "numpy.ndarray":
             "by Pillow if Bits Allocated = 8"
         )
 
+    photometric_interpretation = cast(str, ds.PhotometricInterpretation)
+    rows = cast(int, ds.Rows)
+    columns = cast(int, ds.Columns)
+    bits_stored = cast(int, ds.BitsStored)
+    bits_allocated = cast(int, ds.BitsAllocated)
+
     pixel_bytes = bytearray()
```

Comment:
It's unfortunate that so much boilerplate casting is needed - is there no other way?   I have little 'typing' experience, but I assume we cannot indicate these in some other way because of the "magic" methods yielding types only at run-time?  Or is there a way to indicate that `Dataset.Rows` (et al) is always an int, in one place?

Reply:
I've been thinking about it, and one way may be to create a stub file with all the possible types for each keyword based off the VR and VM from the data dictionary.

But there'd still be a some unavoidable boilerplate casting because even elements with VM 1 may be Type 2 or 3 and hence be `Optional[T]`. The rest would be `Union[None, T, MutableSequence[T]]`

Reply:
Another pie in the sky idea might be to create something similar to [typeshed](https://github.com/python/typeshed), but for each SOP Class. We'd need code to be able to parse the module tables in PS3.3 though. I'm not really sure the work required justifies the benefits (at least for us).